providers/azure.go

```diff
@@ -230,53 +323,32 @@ func getEmailFromJSON(json *simplejson.Json) (string, error) {
 		err = otherMailsErr
 	}
 
+	if err != nil || email == "" {
+		email, err = json.Get("userPrincipalName").String()
```

Comment:
I think this is fine to keep for this PR since the pattern already existed, but in case you want to remedy the design while you are extending it.

In cases where the JSON schema response is static, we should use a struct with JSON tags.

We really only use `simplejson` when fields in the JSON response are configurable by the user with config flags.

Reply:
i will leave it as is for now. really need to wrap this up :)